hitman3.asl

```diff
@@ -1,58 +1,356 @@
+        {"LOCATION_BANGKOK_ZIKA", "level4"},
+        {"LOCATION_BANGKOK", "level4"},
+        {"LOCATION_COLORADO_RABIES", "level5"},
+        {"LOCATION_COLORADO", "level5"},
+        {"LOCATION_HOKKAIDO_FLU", "level6"},
+        {"LOCATION_HOKKAIDO", "level6"},
+        {"LOCATION_NEWZEALAND", "level7"},
+        {"LOCATION_MIAMI", "level8"},
+        {"LOCATION_MIAMI_COTTONMOUTH", "level8"},
+        {"LOCATION_COLOMBIA_ANACONDA", "level9"},
+        {"LOCATION_COLOMBIA", "level9"},
+        {"LOCATION_MUMBAI_KINGCOBRA", "level10"},
+        {"LOCATION_MUMBAI", "level10"},
+        {"LOCATION_NORTHAMERICA", "level11"},
+        {"LOCATION_NORTHAMERICA_GARTERSNAKE", "level11"},
+        {"LOCATION_NORTHSEA", "level12"},
+        {"LOCATION_GREEDY_RACCOON", "level13"},
+        {"LOCATION_OPULENT_STINGRAY", "level14"},
+        {"LOCATION_GOLDEN_GECKO", "level15"},
+        {"LOCATION_GOLDEN", "level15"},
+        {"LOCATION_ANCESTRAL_BULLDOG", "level16"},
+        {"LOCATION_ANCESTRAL_SMOOTHSNAKE", "level16"},
+        {"LOCATION_ANCESTRAL", "level16"},
+        {"LOCATION_EDGY_FOX", "level17"},
+        {"LOCATION_EDGY", "level17"},
+        {"LOCATION_WET_RAT", "level18"},
+        {"LOCATION_WET", "level18"},
+        {"LOCATION_ELEGANT_LLAMA", "level19"},
+        {"LOCATION_ELEGANT", "level19"},
+        {"LOCATION_TRAPPED_WOLVERINE", "level20"},
+        {"LOCATION_TRAPPED", "level20"}
+    };
 }
 
-startup {
-    print("[hitman asl] startup");
-    vars.totalIGT = 0.0;
-    refreshRate = 30;
+start
+{
+    if (settings.StartEnabled && settings["autorestart"])
+    {
+        if (vars.currentMissionTime == 0)
+        {
+            string settingKey = string.Empty;
+            if (vars.locationDescriptor.TryGetValue(vars.currentMetadataLocation, out settingKey))
+            {
+                if (settings[settingKey])
+                {
+                    vars.totalIGT = (ulong)0;
+                    return true;
+                }
+            }
+        }
+    }
+    else
+    {
+        if (vars.totalIGT > 0)
+        {
+            vars.totalIGT = (ulong)0;
+        }
+    }
+
+    return false;
 }
 
-start {
-    if (vars.totalIGT > 0) {
-        print("[hitman asl] livesplit reset");
-        vars.totalIGT = 0.0;
+reset
+{
+    if (vars.currentMissionTime > 0)
+    {
+        vars.disableReset = false;
+    }
+
+    if (settings.ResetEnabled
+        && vars.disableReset == false
+        && settings["autorestart"]
+        && vars.currentGateExited == false
+        && vars.currentMissionTime == 0
+        && vars.currentHudMissionTimer > 0)
+    {
+        string settingKey = string.Empty;
+        if (vars.locationDescriptor.TryGetValue(vars.currentMetadataLocation, out settingKey))
+        {
+            if (settings[settingKey])
+            {
+                vars.disableReset = true;
+                return true;
+            }
+        }
     }
+
+    return false;
+}
+
+split
```

Comment:
I guess it makes sense to put this at the bottom to make it more clear that it runs after gameTime